### ISSUE TITLE
DataTile sources behave unexpectedly when wrapping

### ISSUE THREAD

**Issue by mike-000** [openlayers/openlayers#13590]
**Describe the bug**
Wrapping behavior with DataTile/GeoTIFF sources is inconsistent with other tile sources.

**To Reproduce**
Specify `wrapX: true` on a DataTile or GeoTIFF source in a global projection.  For example https://codesandbox.io/s/cog-stretch-forked-pd8hrn?file=/main.js the real world appears with empty worlds on either side,  In the Data Tile example the displayed x coordinates range from -Infinity to +Infinity instead of repeating as in the Canvas Tiles example when that is wrapped.

**Expected behavior**
If the source calls the loader with normal world tile coordinates as in https://codesandbox.io/s/cog-stretch-forked-xw9w1g?file=/main.js the world will wrap as expected..

**Comment by xmsuperman** [comment]
When initializing the map, set the options of the view, specify the project property as "EPSG: 4326", and the tileNumber returned by the loader method of DataTile: X, Y, Z is incorrect.

` view: new View({          
          projection: "EPSG:4326", 
          center: [0, 0],
          zoom: 0, 
        }),`

However, when manually specifying the initialization of the Projection object for EPSG: 4326 and assigning it to the view, the tileNubmer returned by the loader method of DataTile is correct.

`
      const extent = [-180, -90, 180, 90];
      let projection = new Projection({
        code: 'EPSG:4326',
        extent: extent,
        worldExtent: extent,
        global: true
      });

      this.map = new Map({
        target: "map",
        layers: [
        ],
        view: new View({
          projection: projection, 
          center: [0, 0],
          zoom: 0,           
        }),
      });
`

This issue can result in DataTile subclasses not responding correctly to the warpX: true setting when using projectlike characters to initialize the view's coordinate system with "EPSG: 4326".



This issue cannot be reproduced at EPSG: 3857.

![incorrect](https://github.com/openlayers/openlayers/assets/32294479/f2c0c905-db14-4585-a7c0-5377ae4253e4)
![correct](https://github.com/openlayers/openlayers/assets/32294479/a8353594-4d2e-4a52-9afe-d5ecca0300f6)

**Comment by mike-000** [comment]
Yes, reprojection always uses tiles from the "real" world, so it wraps as expected.

**Comment by xmsuperman** [comment]
So, what should I do, that is, to maintain 'have the user transforms to/from EPSG:3857' and to use the correct xyz TileNum in 'EPSG:4326'.
